x/mongo/driver/operation.go

```diff
@@ -400,6 +400,16 @@ func (op Operation) Execute(ctx context.Context, scratch []byte) error {
 				return err
 			}
 			defer conn.Close()
+
+			// Set the server if the "session.Client" instance deferred the implicit session checkout.
```

Comment:
```suggestion
			// Set the server if it has not already been set and the session type is implicit.
```
The `if` check here is not _really_ checking if the `session.Client` instance deferred the implicit session checkout.